### PR TITLE
Fix/#1091 seasonal doy adjustment

### LINKED ISSUE #1091: BUG: `percentile_doy` does not work when `select_time(data, season="DJF", drop=True)` is applied first.
### Setup Information

- Xclim version: 0.36.3-beta
- Python version: any
- Operating System: any

### Description

When computing percentiles on a subset of the input data where the data of interest is in-between two years,
percentile_doy throws an error:

<details>
<summary>Error stacktrace</summary>

```
Traceback (most recent call last):
  File "/Users/aoun/.conda/envs/xclim/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-16-a61e09936ccd>", line 1, in <module>
    xclim.core.calendar.percentile_doy(x,)
  File "<boltons.funcutils.FunctionBuilder-0>", line 2, in percentile_doy
  File "/Users/aoun/workspace/xclim/xclim/core/formatting.py", line 360, in _call_and_add_history
    outs = func(*args, **kwargs)
  File "/Users/aoun/workspace/xclim/xclim/core/calendar.py", line 520, in percentile_doy
    p = adjust_doy_calendar(p.sel(dayofyear=(p.dayofyear < 366)), arr)
  File "/Users/aoun/workspace/xclim/xclim/core/calendar.py", line 670, in adjust_doy_calendar
    return _interpolate_doy_calendar(source, doy_max)
  File "/Users/aoun/workspace/xclim/xclim/core/calendar.py", line 636, in _interpolate_doy_calendar
    filled_na.coords["dayofyear"] = np.linspace(
  File "/Users/aoun/.conda/envs/xclim/lib/python3.9/site-packages/xarray/core/coordinates.py", line 41, in __setitem__
    self.update({key: value})
  File "/Users/aoun/.conda/envs/xclim/lib/python3.9/site-packages/xarray/core/coordinates.py", line 166, in update
    self._update_coords(coords, indexes)
  File "/Users/aoun/.conda/envs/xclim/lib/python3.9/site-packages/xarray/core/coordinates.py", line 342, in _update_coords
    dims = calculate_dimensions(coords_plus_data)
  File "/Users/aoun/.conda/envs/xclim/lib/python3.9/site-packages/xarray/core/dataset.py", line 204, in calculate_dimensions
    raise ValueError(
ValueError: conflicting sizes for dimension 'dayofyear': length 91 on <this-array> and length 365 on {'lon': 'lon', 'lat': 'lat', 'dayofyear': 'dayofyear', 'percentiles': 'percentiles'}
```
</details>

The error is thrown specifically when the season is between two years because of the 366th day interpolation in percentile_doy:
```
 if p.dayofyear.max() == 366:
        p = adjust_doy_calendar(p.sel(dayofyear=(p.dayofyear < 366)), arr)
```

### Steps To Reproduce

```
def per_doy_season():
    da = xr.open_dataset(
        "/Users/aoun/workspace/xclim/climpact.sampledata.gridded.1991-2010.nc").tmax
    dada= xclim.core.calendar.select_time(da, date_bounds=("07-19", "08-14"), drop=True)
    res = xclim.core.calendar.percentile_doy(dada)

per_doy_season()
```

### Additional context

I don't think this is a high priority bug because setting `drop` to False can workaround this issue.

### Contribution

- [x] I would be willing/able to open a Pull Request to address this bug.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bzah**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1091 #1097 
- [x] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [x] `bumpversion patch` has been called on this branch
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?
This PR makes adjust_doy_calendar works with partially filled time-series such as seasonal time series.
It also fix a bug on missing checks, where `drop=True` indexer kwargs was not properly overriding the current indexer.

### Does this PR introduce a breaking change?
No.

### CODE REVIEW

File: xclim/testing/tests/test_calendar.py [hunk 1]
```diff
@@ -147,6 +147,51 @@ def test_adjust_doy_360_to_366():
...
+    out = adjust_doy_calendar(source, target)
+    # THEN
+    assert out[0].dayofyear == 153
+    assert out[0] == source[0]
+    assert out[-1].dayofyear == 244
+    assert out[-1] == source[-1]
+
+
+# def test_adjust_doy__leap_to_noleap_djf():
+#     # FIXME:
```
Comment (bzah) [inline xclim/testing/tests/test_calendar.py:165]: I'm unsure how to handle partial time-series in-between two years, such as DJF case.

**Comment by bzah** [comment]
@Zeitsperre should I bump patch number again ? It's not clear to me if you want it done on every PR.

**Comment by bzah** [comment]
@Zeitsperre should I bump patch number again ? It's not clear to me if you want it done on every PR.

**Comment by Zeitsperre** [comment]
@bzah It helps with keeping track of the potential introduction of bugs on the main branch. I say bump it. Thanks!

**Comment by bzah** [comment]
@Zeitsperre I'm wondering if this should be part of the CI. 
I think it would not be too hard to make the CI call `bumpversion patch` whenever the pull-request is opened or when master is merged to it. 
What do you think ?

(I just had a look and this would be run when event is `pull_request opened` or ``pull_request synchronized`. For the the later it would also be necessary to check that HEAD is on a merge commit).

**Pull request merged**